rocky/reports/report_types/vulnerability_report/report.py

```diff
@@ -62,15 +61,15 @@ def collect_data(self, input_oois: Iterable[Reference], valid_time: datetime) ->
                         if finding.finding_type.tokenized.id != finding_type.id:
                             continue
 
-                        if finding.primary_key not in history_cache:
+                        if finding.reference not in history_cache:
                             history_cache[finding.reference] = self.octopoes_api_connector.get_history(
                                 finding.reference
                             )
 
-                        time_history = [transaction.valid_time for transaction in history_cache[finding.reference]]
-
-                        if time_history:
-                            first_seen = str(time_history[0])
+                        if history_cache[finding.reference]:
+                            first_seen = str(history_cache[finding.reference][0].valid_time)
```

Comment:
While the string representation of the `valid_time` attribute yields the same result, using the `isoformat()` method is more explicit (and imo appropriate) if the attribute is a `datetime` object. I'll leave it up to you